sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -167,48 +118,33 @@ object ScalaNativePluginInternal {
       }
       libs
     },
+    nativeLinkingOptions in Compile := nativeLinkingOptions.value,
+    nativeLinkingOptions in Test := nativeLinkingOptions.value,
     nativeMode := "debug",
+    nativeMode in Compile := nativeMode.value,
+    nativeMode in Test := nativeMode.value,
     nativeLinkerReporter := tools.LinkerReporter.empty,
+    nativeLinkerReporter in Compile := nativeLinkerReporter.value,
+    nativeLinkerReporter in Test := nativeLinkerReporter.value,
     nativeOptimizerReporter := tools.OptimizerReporter.empty,
-    nativeLogger := streams.value.log,
-    nativeGC := "boehm"
+    nativeOptimizerReporter in Compile := nativeOptimizerReporter.value,
+    nativeOptimizerReporter in Test := nativeOptimizerReporter.value,
+    nativeGC := "boehm",
+    nativeGC in Compile := nativeGC.value,
+    nativeGC in Test := nativeGC.value
   )
 
-  lazy val testSettings = Seq(
-    test := (test in NativeTest).value,
-    testOnly := (testOnly in NativeTest).evaluated,
-    testQuick := (testQuick in NativeTest).evaluated
-  )
-
-  lazy val nativeTestSettings =
-    Defaults.compileSettings ++
-      Defaults.testSettings ++
-      Seq(
-        classDirectory := (classDirectory in Test).value,
-        dependencyClasspath := (dependencyClasspath in Test).value,
-        parallelExecution in test := false,
-        sourceGenerators += Def.task {
-          val frameworks = (loadedTestFrameworks in Test).value.map(_._2).toSeq
-          val tests      = (definedTests in Test).value
-          val output     = sourceManaged.value / "FrameworksMap.scala"
-          IO.write(output, TestUtilities.makeTestMain(frameworks, tests))
-          Seq(output)
-        }.taskValue,
-        loadedTestFrameworks := {
-          val frameworks = (loadedTestFrameworks in Test).value
-          val logger     = streams.value.log
-          val testBinary = nativeLink.value
-          val envVars    = (Keys.envVars in (Test, test)).value
-          (frameworks.zipWithIndex).map {
-            case ((tf, f), id) =>
-              (tf,
-               new ScalaNativeFramework(f, id, logger, testBinary, envVars))
-          }
-        },
-        definedTests := (definedTests in Test).value
-      )
-
-  lazy val scopedSettings = Seq(
+  lazy val scalaNativeConfigSettings: Seq[Setting[_]] = Seq(
+    nativeWarnOldJVM := {
+      val logger = nativeLogger.value
+      Try(Class.forName("java.util.function.Function")).toOption match {
+        case None =>
+          logger.warn("Scala Native is only supported on Java 8 or newer.")
+        case Some(_) =>
+          ()
+      }
+    },
+    nativeLogger := streams.value.log,
```

Comment:
This `nativeLogger` is bullshit. You must directly use `streams.value.log` in tasks that require a logger. Otherwise `last` will not work for those tasks.

`streams` is magical: it gives you a task-specific `Logger` depending on which task requests it. Right now you are bypassing that mechanism, and all logs are associated with the `nativeLogger` task instead of the relevant tasks.